src/Core/Content/Cms/SalesChannel/SalesChannelCmsPageLoader.php

```diff
@@ -204,4 +201,41 @@ private function extractProductIds(EntityCollection $pages): array
             ...array_map(EntityCacheKeyGenerator::buildCmsTag(...), $pages->getIds()),
         ];
     }
+
+    /**
+     * Recursively merges two arrays, with values from the override array taking precedence.
+     * For associative arrays, merges recursively. For indexed arrays, replaces completely.
+     */
+    private function overrideArray(array $original, array $override): array
+    {
+        foreach ($override as $key => $value) {
+            if (\array_key_exists($key, $original) && \is_array($original[$key]) && \is_array($value)) {
```

Comment:
```suggestion
                if (
                    $originalValue !== null && 
                    is_array($originalValue) && 
                    is_array($value) && 
                    !array_is_list($originalValue) && 
                    !array_is_list($value)
                ) {
                    $original[$key] = $this->overrideArray($originalValue, $value);
                } else {
                    $original[$key] = $value;
                }
            
```

The `else` statements do the same, so need need for nested ifs